runtime/ts/type-variable.ts

```diff
@@ -7,13 +7,18 @@
 // http://polymer.github.io/PATENTS.txt
 'use strict';
 
-import {Type} from './ts-build/type.js';
-import {assert} from '../platform/assert-web.js';
-import {Schema} from './ts-build/schema.js';
+import {Type} from './type.js';
+import {assert} from '../../platform/assert-web.js';
+import {Schema} from './schema.js';
 
 export class TypeVariable {
-  constructor(name, canWriteSuperset, canReadSubset) {
-    assert(typeof name == 'string');
+  name: string;
+  _canWriteSuperset: Type|null;
+  _canReadSubset: Type|null;
+  _resolution: Type|null;
+
+  constructor(name: string, canWriteSuperset: Type|null, canReadSubset: Type|null) {
+    assert(typeof name === 'string');
```

Comment:
These asserts seem a bit redundant; can we remove them now?

Reply:
done